### PR TITLE
[runtime] add the ability to specify json config to use inside the script

### PR DESCRIPTION
**Pull request by Ramzeeset**
# Add the ability to specify json config to use inside the script

specifying
--------
It is now possible to specify a special json congifuration when starting the server, which will be available from runtime
```
./server --runtime-config /path/to/config.json
```

Usage
--------
```php
...
$config = kphp_runtime_config();
echo $config["tmp"];
...
```

Some important things
------------------------------

* Changing the received config is valid only until the end of the current script and does not affect subsequent scripts and other workers
* If no config was specified at start, the function will return null

### CODE REVIEW

File: server/php-engine.cpp [hunk 1]
```diff
@@ -2245,6 +2263,7 @@ void parse_main_args(int argc, char *argv[]) {
                                                                                              "memory limit = per_process_memory * processes_count");
   parse_option("job-workers-shared-messages-process-multiplier", required_argument, 2031, "Coefficient used to calculate the total count of the shared messages for job workers related communication:\n"
                                                                                           "messages count = coefficient * processes_count");
+  parse_option("runtime-congifuration", required_argument, 2032, "Smart text"); // add option for runtime
```
Comment (quasilyte) [inline server/php-engine.cpp:2266]: What's "smart text"?

Comment (Ramzeeset) [reply server/php-engine.cpp:2266]: oyyyyyy

Comment (Ramzeeset) [reply server/php-engine.cpp]: fix

**Review by DrDet — COMMENTED** [review]
Let's add python tests for the following cases:
- read only normal usage
- write usage to check that copy-on-write to script memory is working correctly

File: server/php-engine.cpp [hunk 2]
```diff
@@ -2143,6 +2145,22 @@ int main_args_handler(int i, const char *long_option) {
...
+        return -1;
+      }
+      std::stringstream stringstream;
+      stringstream << file.rdbuf();
+      auto [config, success] = json_decode(string(stringstream.str().c_str()));
+      if (!success) {
+        kprintf("--%s option : file is not json\n", long_option);
+        return -1;
+      }
+      runtime_config = config;
```
[the remark this answers is not in the document]
Comment (Ramzeeset) [reply server/php-engine.cpp]: fix

File: server/php-engine.cpp [hunk 3]
```diff
@@ -2245,6 +2263,7 @@ void parse_main_args(int argc, char *argv[]) {
                                                                                              "memory limit = per_process_memory * processes_count");
   parse_option("job-workers-shared-messages-process-multiplier", required_argument, 2031, "Coefficient used to calculate the total count of the shared messages for job workers related communication:\n"
                                                                                           "messages count = coefficient * processes_count");
+  parse_option("runtime-configuration", required_argument, 2032, "name of the json file that will be available at runtime");
```
Comment (DrDet) [inline server/php-engine.cpp:2266]:
```suggestion
  parse_option("runtime-config", required_argument, 2032, "JSON file path that will be available at runtime as 'mixed' via 'kphp_runtime_config()'");
```

File: tests/python/tests/http_server/test_runtime_config.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+
+    def test_read_only(self):
+        response = self.kphp_server.http_request(uri="/test_runtime_config?mode=read_only")
+        self.assertEqual(response.content, b'name : simple configuration version : 1.0.0 OK')
+
+    def test_modifying(self):
+        response = self.kphp_server.http_request(uri="/test_runtime_config?mode=modify")
+        print(response.content)
+        self.assertEqual(response.content, b'modified version : 1.0.1 old version : 1.0.0 OK')
```
Comment (DrDet) [inline tests/python/tests/http_server/test_runtime_config.py:19]: Let's assert that response code is 200 OK

File: tests/python/tests/http_server/php/data/runtime_configuration.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "simple configuration",
+  "version": "1.0.0"
+}
```
Comment (DrDet) [inline tests/python/tests/http_server/php/data/runtime_configuration.json:4]: nit: new line, pls

File: tests/python/tests/http_server/test_runtime_config.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        })
+
+
+    def test_read_only(self):
+        response = self.kphp_server.http_request(uri="/test_runtime_config?mode=read_only")
+        self.assertEqual(response.content, b'name : simple configuration version : 1.0.0 OK')
+
+    def test_modifying(self):
+        response = self.kphp_server.http_request(uri="/test_runtime_config?mode=modify")
+        print(response.content)
```
Comment (DrDet) [inline tests/python/tests/http_server/test_runtime_config.py:18]: redundant print

File: server/php-engine.cpp (hunk 3 above)
Comment (Ramzeeset) [reply server/php-engine.cpp]: fix

File: tests/python/tests/http_server/test_runtime_config.py (hunk 4 above)
Comment (Ramzeeset) [reply tests/python/tests/http_server/test_runtime_config.py]: fix

**Review by DrDet — DISMISSED** [review]

**Pull request merged**